frontend/src/metabase/query_builder/components/notebook/steps/JoinStep/JoinTablePicker/JoinTablePicker.tsx

```diff
@@ -106,12 +112,12 @@ export function JoinTablePicker({
         selectedTableId={tableId}
         setSourceTableFn={handleTableChange}
         triggerElement={
-          <PickerButton disabled={readOnly}>
+          <PickerButton disabled={!canChangeTable}>
```

Comment:
This is the actual fix. The rest of the code is mostly needed to remove the "highlight" effect on the notebook cell's hover